src/main/java/seedu/duke/command/AddNoteCommand.java

```diff
@@ -39,23 +42,43 @@ public AddNoteCommand(Note note) {
 
     @Override
     public String execute() {
+        String content = note.getContent();
+        boolean isInputSuccess = false;
+
         // Search for duplicates
         ArrayList<Note> filteredTaskList = (ArrayList<Note>) notebook.getNotes().stream()
-                .filter((s) -> s.getTitle().equals(note.getTitle()))
+                .filter((s) -> s.getTitle().equalsIgnoreCase(note.getTitle()))
                 .collect(toList());
 
         if (!filteredTaskList.isEmpty()) {
-            return COMMAND_UNSUCCESSFUL_MESSAGE;
+            return Formatter.formatString(COMMAND_UNSUCCESSFUL_MESSAGE, false);
+        }
+
+        // Get Content
+        if (content.isBlank()) {
+            do {
+                System.out.println("Enter Note:");
+                try {
+                    content = inputContent();
+                    isInputSuccess = true;
+                } catch (StringIndexOutOfBoundsException exception) {
+                    System.out.println(SystemException.ExceptionType.EXCEPTION_INVALID_END_INPUT);
+                }
+            } while (!isInputSuccess);
         }
```

Comment:
Can refactor most of this into inputContent(), including the do while.

Eventually it should just be:

content = inputContent;

if (content == null) {
return Formatter.formatMessage(unsuccessful message);
} else {
note.setContent(content);
...
...
...
return Formatter.formatMessage(successful message +...)
}

Reply:
Feels like a little bit of arrowhead coding. Jun Xiang has a good suggestion to reduce it.